clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6861,13 +6861,19 @@ ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D, const ParsedAttr &AL,
   // though they were unknown attributes.
   if (AL.getKind() == ParsedAttr::UnknownAttribute ||
       !AL.existsInTarget(S.Context.getTargetInfo())) {
-    S.Diag(AL.getLoc(),
-           AL.isRegularKeywordAttribute()
-               ? (unsigned)diag::err_keyword_not_supported_on_target
-           : AL.isDeclspecAttribute()
-               ? (unsigned)diag::warn_unhandled_ms_attribute_ignored
-               : (unsigned)diag::warn_unknown_attribute_ignored)
-        << AL << AL.getRange();
+    auto D = S.Diag(AL.getLoc(),
+                    AL.isRegularKeywordAttribute()
+                        ? (unsigned)diag::err_keyword_not_supported_on_target
+                    : AL.isDeclspecAttribute()
+                        ? (unsigned)diag::warn_unhandled_ms_attribute_ignored
+                        : (unsigned)diag::warn_unknown_attribute_ignored);
```

Comment:
```suggestion
                        ? diag::err_keyword_not_supported_on_target
                    : AL.isDeclspecAttribute()
                        ? diag::warn_unhandled_ms_attribute_ignored
                        : diag::warn_unknown_attribute_ignored);
```
We can get rid of the casts, can't we?

Reply:
Sure. I've removed. Thanks